julia/core.py

```diff
@@ -46,7 +46,45 @@ class JuliaError(Exception):
 
 
 class JuliaModule(ModuleType):
-    pass
+
+    def __init__(self, loader, *args, **kwargs):
+        super(JuliaModule, self).__init__(*args, **kwargs)
+        self.__julia = loader.julia
+        self.__loader__ = loader
+
+    @property
+    def __all__(self):
+        juliapath = self.__name__.lstrip("julia.")
+        names = set(self.__julia.eval("names({})".format(juliapath)))
+        names.discard(juliapath.rsplit('.', 1)[-1])
+        return list(names)
+
+    def __getattr__(self, name):
+        juliapath = self.__name__.lstrip("julia.")
+        attrname = name
+        if name.endswith("!"):
+            attrname = name.replace("!", "_b")
+        if keyword.iskeyword(name):
+            attrname = "jl".join(name)
+        try:
+            module_path = ".".join((juliapath, name))
+            is_module = self.__julia.eval("isa({}, Module)"
+                                          .format(module_path))
+            if is_module:
+                # TODO: find a better way to handle submodules
+                split_path = module_path.split(".")
+                is_base = split_path[-1] == "Base"
+                recur_module = split_path[-1] == split_path[-2]
+                if not is_base and not recur_module:
```

Comment:
Not sure what is the best way to handle submodules.  Is there an easy way to check if some module is a submodule of other module in Julia?  If not, it's probably better to let user explicitly import submodules (i.e., `julia.Module1.Module2` is an `AttributeError` until the user does `import julia.Module1.Module2`)?